### PR TITLE
[master] Add rename popup for extract to const and local var

### LINKED ISSUE #37479: Add rename support for extract refactoring code actions
**Description:**
Add rename support to `extract to constant` and `extract to local variable` code actions.

**Related Issues (optional):**
https://github.com/ballerina-platform/ballerina-lang/issues/36568
https://github.com/ballerina-platform/ballerina-lang/issues/36569

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
This PR will add rename popup for the extract to constant and extract to local variable code actions.

Fixes #37479
Related PRS #37891(2201.2.x-closed), #38225(2201.3.x)

https://user-images.githubusercontent.com/46857198/192220255-a6301a6a-85f5-4af1-be98-9fc67697ff41.mov

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConstantCodeAction.java [hunk 1]
```diff
@@ -128,14 +135,29 @@ private String getLocalVarName(CodeActionContext context) {
...
         ImportDeclarationNode lastImport = importsList.get(importsList.size() - 1);
         return new Position(lastImport.lineRange().endLine().line() + 2, 0);
     }
 
+    private void addRenamePopup(CodeActionContext context, CodeAction codeAction,
+                                int constDeclStrLength, int nodeStartOffset) {
+        Optional<SyntaxTree> syntaxTree = context.currentSyntaxTree();
+        if (syntaxTree.isEmpty()) {
+            return;
+        }
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConstantCodeAction.java:151]: Why do we have this check here?

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/variableAssignmentRequiredCodeAction7.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
         "command": "ballerina.action.rename",
         "arguments": [
           "createVariable2.bal",
-          494
+          492
```
[the remark this answers is not in the document]
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/variableAssignmentRequiredCodeAction7.json:31]: yes

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConstantCodeAction.java (hunk 1 above)
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConstantCodeAction.java:151]: Removed this in, both ExtractToConstant and ExtractToLocalVariables code actions.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/constants/CommandConstants.java [hunk 3]
```diff
@@ -177,6 +178,12 @@ public class CommandConstants {
     public static final String EXTRACT_TO_CONSTANT = "Extract to constant";
 
     public static final String EXTRACT_TO_VARIABLE = "Extract to local variable";
+    
+    public static final String BALLERINA_RENAME_ACTION = "ballerina.action.rename";
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/constants/CommandConstants.java:182]: Is this the best place we can put this? @IMS94 @malinthar

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java [hunk 4]
```diff
@@ -59,7 +61,7 @@ public class ExtractToLocalVarCodeAction implements RangeBasedCodeActionProvider
 
     public static final String NAME = "Extract To Local Variable";
     private static final String VARIABLE_NAME_PREFIX = "var";
-
+
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java:64]: Is this extra tab intentional? is it from the checkstyles?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/constants/CommandConstants.java (hunk 3 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/constants/CommandConstants.java:182]: `ballerina.action.rename` is a command residing in the vscode plugin side. So, I'm ok with the place. Only issue is the name of the variable. Shall we name it `RENAME_COMMAND`?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java (hunk 4 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java:64]: That's probably when we apply `ctrl + alt +L` to format the entire file. @LakshanWeerasinghe you are using the WSO2 coding styles right?

Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java:64]: @IMS94 yes I am

Comment (gayalkuruppu) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java:64]: @IMS94 yes, I have used that. but haven't come across indentations after constants though. Are there any general style practices like this?

Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToLocalVarCodeAction.java:64]: Although I haven't select `keep indents on empty lines`, intellij indents empty lines. Do I need to make this change? @IMS94

**Comment by IMS94** [comment]
@LakshanWeerasinghe can you check the test failures on windows? May be related to line separator character difference.

**Pull request closed without merging**